game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -45,8 +46,27 @@ class ResourceType(grid:Grid, val min: Int, val max: Int, mass: Int, score: Int)
 
   def spawnResources: Unit = {
     val ratio = (positions.length - min).toFloat / (max - min)
-    if (scala.util.Random.nextFloat >= ratio) {
+    if ((scala.util.Random.nextFloat + 0.000001) >= ratio) {
       positions :::= List(grid.randomPosition)
     }
   }
-}
\ No newline at end of file
+
+  /** Small loop to detect if a cell can eat a resource **/
+  def detectCollision(players: List[Player]): Unit = {
+    for(player <- players) {
+      for(cell <- player.cells) {
+        for(position <- positions){
+          if(cell.contains(position)){
+            reward(player, cell, mass, score)
+            /** Returns a new list without the resource that has been consumed **/
+            positions = positions.filterNot(a => a == position)
+          }
+        }
+      }
+    }
+  }
+
+  def reward(player: Player, cell: Cell, mass: Int, score: Int): Unit = {
```

Comment:
Forgot to remove it after I realized I wouldn't increase score in this PR. I will remove it.